compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingBase.td

```diff
@@ -88,7 +74,7 @@ def EncodingAttr :
   );
 
   let builders = [
-    AttrBuilder<(ins "EncodingRole":$role,
+    AttrBuilder<(ins "int64_t":$role,
```

Comment:
Please consistently replace `role` by `operandIndex` everywhere.

Reply:
Done. I changed all "role" to operand index in IREE. 

Checking other MLIR impl and they indeed use "role" to represent LHS, RHS, etc. So Indeed "role" is the term used in matmul. Now we are extending it outside of matmul and use index instead, would that cause confusions for someone who are familiar with matmul optimizations from outside?

Reply:
Can you share a link to the MLIR code you were looking at ?

Reply:
Sure, [here](https://github.com/shark-infra/llvm-project/blob/cafb6284d18bbdb952ae6d5e4aa97912d57dbfb8/mlir/include/mlir/Dialect/NVGPU/Utils/MMAUtils.h#L26)

Reply:
I feel that it is nvgpu things, but not linalg.

<img width="732" alt="image" src="https://github.com/iree-org/iree/assets/6949191/9e3a85ba-8fee-428a-afa2-b2ece843f469">

I don't find any `role` terms in linalg codes.

<img width="790" alt="image" src="https://github.com/iree-org/iree/assets/6949191/ee997c26-40a3-4836-b0e8-08a4b87a6aac">

Reply:
coincidence or people just use the term for such cases?